semgrep/semgrep/semgrep_types.py

```diff
@@ -157,3 +159,9 @@ def __eq__(self, other: Any) -> bool:
             and self.end == other.end
             and self.vars_match(other)
         )
+
+    def __lt__(self, other: Any) -> bool:
```

Comment:
This was required for sorting. I'm not sure which makes more sense here: 1.) "less than" means the start and end are smaller, or 2.) "less than" means end - start is smaller. Either way, it doesn't really matter, we just need to sort to get predictable results for comparison.

Reply:
Do we need to break ties for stability or is it fine?

Reply:
Ahhh, good call! I was banging my head against a wall all Friday wondering why I kept getting different results on Python 3.7 vs. 3.8. Looks like it's because there's no stable tie breaker, like you mentioned :+1: 

I'll add one - I think (end - start) with tie breaks going to smaller line numbers should do the trick.